go.mod

```diff
@@ -24,3 +25,5 @@ require (
 replace google.golang.org/grpc => google.golang.org/grpc v1.33.2
 
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+
+replace github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
```

Comment:
if we expect more dashes in the chain id, then perhaps using this:
https://github.com/crypto-org-chain/keyring/commits/v1.1.6-fixes which includes @leejw51crypto 's patch for the linux secret service in keyring

Reply:
extracted the keyring change to standalone PR.